packages/babel-plugin/src/utils/ast-builders.tsx

```diff
@@ -145,6 +145,109 @@ const traverseStyledBinaryExpression = (node: t.BinaryExpression, nestedVisitor:
   return node;
 };
 
+/**
+ * Handles cases like:
+ * 1. `propz.loading` in `border-color: \${(propz) => (propz.loading ? colors.N100 : colors.N200)};`
+ * Outcome: It will replace `propz.loading` with `props.loading`.
+ *
+ * 2. `prop.notValidProp` in `border-color: \${(props) => (props.notValidProp ? colors.N100 : colors.N200)};`
+ * Outcome: It will move `notValidProp` under `propsToDestructure` and replaces `props.notValidProp` with `notValidProp`.
+ *
+ * @param path MemberExpression path
+ * @param propsToDestructure Array which gets mutated with props which are not valid eg. props.notValidProp
+ */
+const handleMemberExpressionInStyledInterpolations = (
+  path: NodePath<t.MemberExpression>,
+  propsToDestructure: string[]
+) => {
+  const memberExpressionKey = path.node.object;
+
+  if (!t.isIdentifier(memberExpressionKey)) {
+    return;
+  }
+
+  const traversedUpFunctionPath = path.find((parentPath) => parentPath.isFunction());
+  const memberExpressionKeyName = memberExpressionKey.name;
+
+  const isMemberExpressionKeyNameEqualsFunctionParams =
+    t.isFunction(traversedUpFunctionPath.node) &&
+    t.isIdentifier(traversedUpFunctionPath.node.params[0]) &&
+    traversedUpFunctionPath.node.params[0].name === memberExpressionKeyName;
+
+  if (isMemberExpressionKeyNameEqualsFunctionParams) {
+    const memberExpressionValue = path.node.property;
+
+    if (!t.isIdentifier(memberExpressionValue)) {
+      return;
+    }
+
+    const memberExpressionValueName = memberExpressionValue.name;
+
+    if (isPropValid(memberExpressionValueName)) {
+      // Convert cases like propz.color to props.color
+      if (memberExpressionKeyName !== 'props') {
```

Comment:
can we move `'props'` to module scope and reuse it? or move to `constants.tsx`? 🤔